tests/base/src/test/java/org/wildfly/security/audit/SyslogAuditEndpointTest.java

```diff
@@ -0,0 +1,331 @@
+    private static final String RFC5424_LOG_MESSAGE = "testing log message RFC5424";
+    private static final int UDP_PORT = 10837;
+    private static final int TCP_PORT = 10857;
+    private static final int RECONNECT_NUMBER = 10;
+    private static final int BAD_RECONNECT_NUMBER = -2;
+    private static final int RECONNECT_TIMEOUT = 10;
+    private static SocketFactory socketFactory = null;
+    private static SimpleSyslogServer udpServer = null;
+    private static SimpleSyslogServer tcpServer = null;
+    private String failString = "";
+
+    /**
+     * Gets the server address and makes the socket factory before starting up the server
+     */
+    @BeforeClass
+    public static void initializeServer() throws Exception {
+        socketFactory = SocketFactory.getDefault();
+        udpServer = SimpleSyslogServer.createUdp(UDP_PORT);
+        tcpServer = SimpleSyslogServer.createTcp(TCP_PORT, false);
+    }
+
+    /**
+     * Safely shutdowns the server
+     */
+    @AfterClass
+    public static void shutdownServer() {
+        udpServer.close();
+        tcpServer.close();
+    }
+
+    /**
+     * Tests the syslog audit endpoint with RFC 3164 format and critical priority
+     */
+    @Test
+    public void testGenericRfc3164SyslogEvent() throws Exception {
+        AuditEndpoint endpoint = setupEndpointUdp(SyslogHandler.SyslogType.RFC3164);
+        endpoint.accept(EventPriority.CRITICAL, RFC3164_LOG_MESSAGE);
+        endpoint.close();
+        discardMessage(udpServer);
+        verifyUdpMessage(RFC3164_LOG_MESSAGE);
+    }
+
+    /**
+     * Tests the syslog audit endpoint with RFC 5424 format and critical priority
+     */
+    @Test
+    public void testGenericRfc5424SyslogEvent() throws Exception {
+        AuditEndpoint endpoint = setupEndpointUdp(SyslogHandler.SyslogType.RFC5424);
+        endpoint.accept(EventPriority.CRITICAL, RFC5424_LOG_MESSAGE);
+        endpoint.close();
+        discardMessage(udpServer);
+        verifyUdpMessage(RFC5424_LOG_MESSAGE);
+    }
+
+    /**
+     * Tests that the endpoint rejects a bad integer
+     */
+    @Test
+    public void testBadIntegerReconnectAttempts() throws Exception {
+        try {
+            // Since the endpoint gives a message upon creation there is no need to call .accept()
+            AuditEndpoint endpoint = setupEndpointBadUdpReconnectAttempts(BAD_RECONNECT_NUMBER);
+            Assert.fail("No error was thrown while attempting to log with an invalid reconnect-attempts number.");
+        } catch (IllegalArgumentException e) {
+            assertCorrectError(e, new String[] {"ELY11008", Integer.toString(BAD_RECONNECT_NUMBER)});
+        }
+    }
+
+    /**
+     * Tests that the server continuously attempts to resend the message for {@link SyslogAuditEndpointTest#RECONNECT_TIMEOUT} milliseconds
+     */
+    @Test
+    public void testFailureInfiniteReconnectAttempts() throws Exception {
```

Comment:
How do you know that it actually attempts to reconnect? Running, exception-less thread is the guarantee?

Reply:
If the endpoint gets successfully created (that is, the initial connection message gets sent), then failString will be set due to the thread finishing without an exception, and if there is an exception thrown during execution then failString will also get set, in either scenario causing the test to fail. Since the only state where the test passes is the child thread waiting for the initial connection message to get sent and then being interrupted by the parent thread, it would guarantee that the endpoint did attempt to continually resend the message.

Reply:
Thank you!